src/__experimental__/components/checkbox/checkbox.style.js

```diff
@@ -50,6 +80,10 @@ const CheckboxStyle = styled.div`
     ${LabelStyle} {
       padding: 0 6px;
       width: auto;
+
+      ${({ hasError }) => hasError && !disabled && css`
```

Comment:
As we're doing it for the rest, probably makes sense to destructure `hasError` on line 39 above.